web: fix turnstile types after update

kensternberg-authentik opened the pull request:
## Details

After running 'npm update' on the dev tree, the build started to fail with these options and types no longer being set correctly in the source tree.

I have explicitly included the Turnstile object as a sub-component of Window, and modified the CaptchaStage to understand the TurnstileObject and TurnstileOptions, and the build now completes.  As these are purely type-level changes, the functionality of the system is unchanged, and testing (by hand, _sigh_) indicates that the visible behavior is consistent with previous iterations.


## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [X] The code has been formatted (`make web`)
-   [ ] The translation files have been updated (`make i18n-extract`)

If applicable

-   [ ] The documentation has been updated
-   [ ] The documentation has been formatted (`make website`)

kensternberg-authentik committed:
web: fix turnstile types after update

After running 'npm update' on the dev tree, the build started to
fail with these options and types no longer being set correctly in
the source tree.

I have explicitly included the Turnstile object as a sub-component
of Window, and modified the CaptchaStage to understand the
TurnstileObject and TurnstileOptions, and the build now completes.

kensternberg-authentik commented on web/src/global.d.ts:
```diff
@@ -1,4 +1,5 @@
 declare module "*.css";
+
```
Prettier _should have_ had an opinion here!  :-)

kensternberg-authentik commented on web/src/flow/stages/captcha/CaptchaStage.ts:
```diff
@@ -110,9 +115,9 @@ export class CaptchaStage extends BaseStage<CaptchaChallenge, CaptchaChallengeRe
+        (window as unknown as TurnstileWindow).turnstile.render(container, {
             sitekey: this.challenge.siteKey,
-            size: "invisible",
+            size: "invisible" as TurnstileOptions["size"],
```
@BeryJu Could you please affirm this?  The `as TurnstilesOptions["size"]` clause was necessary here because "invisible" is not a valid setting according to the options in the provided type file.  If `invisible` is valid because you've made it so, let me know and I'll a comment to that effect.

BeryJu replied:
I don't think thats valid (at least not anymore), so we can probably remove it

kensternberg-authentik committed:
eslint says to prefer this format

kensternberg-authentik committed:
Google recaptcha (aka Turnstile) doesn't understand the "invisible" setting; that's purely
an HCaptcha thing.

kensternberg-authentik committed:
web: removing the typecast means I no longer need the type.

BeryJu reviewed (commented):
As https://github.com/goauthentik/authentik/pull/7797 is still open, could you also lock pyright to the previous version as it's currently breaking CI

kensternberg-authentik committed:
Locking pyright to 1.1.338 and maintaining it.

kensternberg-authentik committed:
web: locking down hard

After reading [this
guide](https://medium.com/@anjusha.khandavalli/decoding-commonly-used-symbols-in-package-json-file-e08f3939c9e4),
I've locked down the version of pyright to a specific and immovable
version until we can get a better read on how the Pyright upgrade
breaks things.

kensternberg-authentik committed:
Update is specific to package-lock.json.

Signed-off-by: Jens Langhammer <jens@goauthentik.io>
